process/elasticproc/elasticProcessor_test.go

```diff
@@ -599,3 +602,120 @@ func TestElasticProcessor_IndexAlteredAccounts(t *testing.T) {
 	require.Nil(t, err)
 	require.True(t, called)
 }
+
+func TestElasticProcessor_DoBulkRequests_SuccessfulParallelProcessing(t *testing.T) {
+	arguments := createMockElasticProcessorArgs()
+	arguments.NumWritesInParallel = 3
+
+	processedBuffers := make(map[string]bool)
+	processedBuffersMutex := &sync.Mutex{}
+
+	arguments.DBClient = &mock.DatabaseWriterStub{
+		DoBulkRequestCalled: func(buff *bytes.Buffer, index string) error {
+			processedBuffersMutex.Lock()
+			processedBuffers[buff.String()] = true
+			processedBuffersMutex.Unlock()
+			return nil
+		},
+	}
+
+	elasticProc, _ := NewElasticProcessor(arguments)
+
+	// Create 5 test buffers
+	buffers := make([]*bytes.Buffer, 5)
+	for i := 0; i < 5; i++ {
+		buffers[i] = bytes.NewBufferString(fmt.Sprintf("test-buffer-%d", i))
+	}
+
+	err := elasticProc.doBulkRequests("test-index", buffers, 0)
+	require.Nil(t, err)
+
+	// Verify all buffers were processed
+	processedBuffersMutex.Lock()
+	defer processedBuffersMutex.Unlock()
+	require.Equal(t, 5, len(processedBuffers))
+	for i := 0; i < 5; i++ {
+		require.True(t, processedBuffers[fmt.Sprintf("test-buffer-%d", i)])
+	}
+}
+
+func TestElasticProcessor_DoBulkRequests_WorkerCancellationOnError(t *testing.T) {
+	arguments := createMockElasticProcessorArgs()
+	arguments.NumWritesInParallel = 3
+
+	processedBuffers := make(map[string]bool)
+	processedBuffersMutex := &sync.Mutex{}
+	expectedErr := errors.New("expected error")
+
+	arguments.DBClient = &mock.DatabaseWriterStub{
+		DoBulkRequestCalled: func(buff *bytes.Buffer, index string) error {
+			processedBuffersMutex.Lock()
+			processedBuffers[buff.String()] = true
+			processedBuffersMutex.Unlock()
+
+			if buff.String() == "test-buffer-1" {
+				return expectedErr
+			}
+			return nil
+		},
+	}
+
+	elasticProc, _ := NewElasticProcessor(arguments)
+
+	buffers := make([]*bytes.Buffer, 5)
+	for i := 0; i < 5; i++ {
+		buffers[i] = bytes.NewBufferString(fmt.Sprintf("test-buffer-%d", i))
+	}
+
+	err := elasticProc.doBulkRequests("test-index", buffers, 0)
+	require.Equal(t, expectedErr, err)
+
+	processedBuffersMutex.Lock()
+	defer processedBuffersMutex.Unlock()
+	require.True(t, len(processedBuffers) > 0)
+	require.True(t, processedBuffers["test-buffer-0"])
+	require.True(t, processedBuffers["test-buffer-1"])
```

Comment:
Why only 0 and 1?  -> please check for all buffers
If you have a `DoBulkRequestCalled` you send err on chan, shouldn't you continue execution?